rust/connlib/snownet/src/node.rs

```diff
@@ -832,6 +848,7 @@ where
             intent_sent_at,
             relay: self.sample_relay(),
             is_failed: false,
+            buffered_packets: RingBuffer::new(10),
```

Comment:
> So while awaiting the connection we are buffering a single packet but while in-connection state we are buffering 10?

`AwaitingConnectionDetails` no longer exists as soon as we have a connection in `snownet`, even if it is just an `InitialConnection`. This buffer may need to be larger than 10 packets, depending on whether we unify them from multiple resources that allow want to connect to the same gateway.

Reply:
I understand and we don't wnat to buffer always in snownet since we want to drop the buffer when the connections drops-- makes sense.

I just don't like having them in different places :( but I think this is a good solution